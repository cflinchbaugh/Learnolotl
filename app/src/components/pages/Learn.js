import React, { Component } from 'react';
import Flashcards from '../organisms/Flashcards';
import Uploader from '../organisms/Uploader';

import { Link } from 'react-router-dom';


class Learn extends Component {
    constructor(props) {
        super(props);

        this.state = {
            cardData: [
                {
                    "metadata": ["101"],
                    "langData": [
                        {
                            "id": "english",
                            "value": "I"
                        }, {
                            "id": "romaji",
                            "value": "watashi"
                        }, {
                            "id": "hiragana",
                            "value": "\u308E\u305F\u3057"
                        }
                    ]
                },
        
                {
                    "metadata": ["101"],
                     "langData": [
                         {
                             "id": "english",
                             "value": "you"
                         }, {
                             "id": "romaji",
                             "value": "anata"
                         }, {
                             "id": "hiragana",
                             "value": "\u3041\u306A\u305F"
                         }
                     ]
                }
            ],
            display: 'none'
        }

        this.handleFileUpload = this.handleFileUpload.bind(this);
    }

    handleFileUpload(updatedCardData) {
         this.setState((state) => ({
            cardData: updatedCardData
        }));
    }


    render() {
        let uploaderData = {
            handleFileUpload: this.handleFileUpload,
            cardData: this.state.cardData
        },
        flashcardsData = {
            cardData: this.state.cardData
        }

        return (
            <div>
                <Uploader {...uploaderData} />
                
                <Link to={{
                    pathname: '/learn/flashcards',
                    state: {...this.state}
                    
                    }}>Flashcards</Link>
            </div>
        );
    }
}

export default Learn;