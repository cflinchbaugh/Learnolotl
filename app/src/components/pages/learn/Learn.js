import React from 'react';

import Uploader from 'organisms/Uploader';
import Select from 'formElements/Select';

function Learn (props) {
    return (
        <div>
            <Uploader {...props.uploaderData} />

            {props.fileListing}

            <Select {...props.modeSelectData} />
            
            
        </div>
    );
}


export default Learn;