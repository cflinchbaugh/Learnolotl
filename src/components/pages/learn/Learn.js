import React from 'react';
import styled from 'styled-components';

import { Link } from 'react-router-dom';
import Uploader from 'learn/Uploader';
import Select from 'formElements/Select';
import NavigationArrow from 'buttons/NavigationArrow';

const StyleWrapper = styled.div`
    margin: 50px auto;

    .navigation-button {
        float: right;
    }

    .uploaded-files-wrapper,
    .uploader-wrapper,
    .mode-select-wrapper {
        margin-bottom: 20px;
    }

    .card-selection {
        text-align: center;
    }
    .link-wrapper {
        margin-left: auto;
        margin-right: 0;
    }
`

function Learn (props) {
    const renderUploader = props.sampleData ? '' : <Uploader {...props.uploaderData} />;
    const renderLink = (!props.sampleData && props.uploadedIds.length) || (props.sampleData) ? 
        (
            <Link to={{
                    pathname: 'Learnolotl/learn/flashcards',
                    state: {
                        mode: props.linkData.mode
                    }    
                }}>
                <NavigationArrow type="next"/>
            </Link>
        ) : '';
    const renderModeSelector = props.sampleData || props.uploadedIds.length ?
        (
            <div className="mode-select-wrapper">
                <Select {...props.modeSelectData} /> 
            </div>
        ) : '';
    const renderUploadedFiles = props.uploadedIds.length ?
        (
            <div className="uploaded-files-wrapper">
                {props.fileListingData.fileListing}
            </div>
        ) : '';

    return (
        <StyleWrapper className="learn-wrapper">
            <div className="card-selection">
                <div className="uploader-wrapper">
                    {renderUploader}
                </div>

                {renderUploadedFiles}
                
                {renderModeSelector}
            
            </div>

            <div className="link-wrapper clearfix">
                {renderLink}
            </div>

        </StyleWrapper>
    );
}


export default Learn;