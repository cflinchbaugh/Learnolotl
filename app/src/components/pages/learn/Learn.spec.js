import React from 'react';
import {shallow} from 'enzyme';
import Learn from './Learn';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('Learn', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const learnData = {
            uploadedIds: ''
        }

        const tree = renderer.render(
            <Learn {...learnData}/>
        );

        expect(tree).toMatchSnapshot();
    });

    
});