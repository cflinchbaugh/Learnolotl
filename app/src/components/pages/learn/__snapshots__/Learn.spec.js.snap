// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn renders and matches snapshot 1`] = `
<styled.div
  className="learn-wrapper"
>
  <div
    className="card-selection"
  >
    <div
      className="uploader-wrapper"
    >
      <Uploader />
    </div>
    
    
  </div>
  <div
    className="link-wrapper clearfix"
  />
</styled.div>
`;
